File: app/src/main/java/org/stepic/droid/ui/fragments/MathStepFragment.java
package org.stepic.droid.ui.fragments;

import org.stepic.droid.model.Reply;

public class MathStepFragment extends SingleLineSendStepFragment {

    @Override
    protected Reply generateReply() {
        return new Reply.Builder()
                .setFormula(answerField.getText().toString())
                .build();
    }

    @Override
    protected void onRestoreSubmission() {
        Reply reply = submission.getReply();
        if (reply == null) return;

        String text = reply.getFormula();
        answerField.setText(text);
    }

}

File: app/src/main/java/org/stepic/droid/ui/fragments/MathStepFragment.kt
package org.stepic.droid.ui.fragments

import org.stepik.android.model.learning.replies.Reply

class MathStepFragment : SingleLineSendStepFragment() {

    override fun generateReply(): Reply = Reply(formula = answerField.text.toString())

    override fun onRestoreSubmission() {
        val formula = submission.reply?.formula ?: return
        answerField.setText(formula)
    }

}
